pkg/cluster/internal/create/actions/installpsp/psp.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package installpsp implements the an action to isntall a default
+// Pod Security Policy
+package installpsp
+
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"sigs.k8s.io/kind/pkg/cluster/internal/create/actions"
+	"sigs.k8s.io/kind/pkg/cluster/nodes"
+)
+
+type action struct{}
+
+// NewAction returns a new action for installing storage
+func NewAction() actions.Action {
+	return &action{}
+}
+
+// Execute runs the action
+func (a *action) Execute(ctx *actions.ActionContext) error {
+	ctx.Status.Start("Installing Default PodSecurityPolicy  💾")
+	defer ctx.Status.End(false)
+
+	allNodes, err := ctx.Nodes()
+	if err != nil {
+		return err
+	}
+
+	// get the target node for this task
+	node, err := nodes.BootstrapControlPlaneNode(allNodes)
+	if err != nil {
+		return err
+	}
+
+	// add the default storage class
+	if err := addDefaultPodSecurityPolicy(node); err != nil {
+		return errors.Wrap(err, "failed to add default Pod Security Policy")
+	}
+
+	// mark success
+	ctx.Status.End(true)
+	return nil
+}
+
+// a default pod security policy unrestricted
+// we need this for e2es (PodSecurityPolicy)
+const defaultPodSecurityPolicyManifest = `# least restricted policy
+---
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: psp-privileged
+  annotations:
+    seccomp.security.alpha.kubernetes.io/allowedProfileNames: '*'
```

Comment:
does it make sense to add the `apparmor... *` here too?
do we have it running on the latest kind node image (Ubuntu)?

Reply:
the problem I had with apparmor is that if you add it and the host doesn't have apparmor it blocks the pod.

Reply:
ok, understood.

Reply:
this part concerns me a bit

Reply:
do we know how widespread this is in EG helm charts?

Reply:
I was digging more into this:
* AKS approach https://github.com/Azure/aks-engine/blob/e492894dfd9bd6668d778c524084254971dd6d88/parts/k8s/addons/kubernetesmasteraddons-pod-security-policy.yaml
* Kops https://github.com/kubernetes/kops/blob/56ccfac26d5a85b72ef9bee7e9c613cd01e749d9/upup/models/cloudup/resources/addons/podsecuritypolicy.addons.k8s.io/k8s-1.10.yaml.template
* kubespray https://github.com/kubernetes-sigs/kubespray/blob/master/roles/kubernetes-apps/cluster_roles/templates/psp.yml.j2
* GCE https://github.com/kubernetes/kubernetes/tree/master/cluster/gce/addons/podsecuritypolicies